### PR TITLE
Add connect sign up to connect

### PR DESCRIPTION
**Pull request by clairelin135**
- Adds connect sign up screen to connect screen if user is not `onConnect`
- Adds sign up screen for connect
- Updates fields of `Veteran` model `military_branch`, `unit`, `notes`, `accept_messages`, `share_profile`, `accept_notices`, `on_connect`
screen shot 2017-11-14 at 11 08 19 pm

### CODE REVIEW

File: src/screens/ConnectSignUpScreen.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    */
+  getFormType() {
+    return t.struct({
+      militaryBranch: t.enums({
+        1: 'Army',
+        2: 'Navy',
+        3: 'Marines',
+        4: 'Air Force',
+        5: 'Coast Guard',
+        6: 'First Responder',
```
Comment (nikitr) [inline src/screens/ConnectSignUpScreen.js]: we should be getting the roles and militaryBranch from the enums defined in the veteran model

File: src/screens/ConnectSignUpScreen.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    this.signUp = this.signUp.bind(this);
+    this.navigateToConnect = this.navigateToConnect.bind(this);
+  }
+
+  async componentDidMount() {
+    try {
+      const endpoint = APIRoutes.veteransMilitaryBranch();
+      let response_json = await BaseRequester.get(endpoint);
+      let militaryBranchDict = {};
+      for(var key in response_json) {
```
Comment (nikitr) [inline src/screens/ConnectSignUpScreen.js]: add space after for

File: src/screens/ConnectSignUpScreen.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        if(response_json.hasOwnProperty(key)) {
```
Comment (nikitr) [inline src/screens/ConnectSignUpScreen.js]: space after if

File: src/screens/ConnectSignUpScreen.js [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    } catch (error) {
+      console.log(error);
+      return Promise.reject(error);
+    }
+  }
+
+  formatDict(str) {
+    newStr = "";
+    for(var i = 0; i < str.length; i++) {
+      if(str[i]=='_') {
```
Comment (nikitr) [inline src/screens/ConnectSignUpScreen.js]: space after for and if

Comment (nikitr) [reply src/screens/ConnectSignUpScreen.js]: and space before and after ==

File: src/helpers/routes/routes.js [hunk 5]
```diff
@@ -13,6 +13,15 @@ class APIRoutes {
...
     return APIRoutes.create('/veterans');
   }
+  static veteransConnectSignUpPath(id) {
+    return APIRoutes.create(`/veterans/${id}/connect_sign_up`);
+  }
+  static veteransConnectStatus(id) {
+    return APIRoutes.create(`/veterans/${id}`);
+  }
+  static veteransMilitaryBranch() {
+    return APIRoutes.create('/veterans/get_military_branch');
```
Comment (nikitr) [inline src/helpers/routes/routes.js:23]: do you have a PR where you define the military_branch enum in the veteran model?

Comment (nikitr) [reply src/helpers/routes/routes.js:23]: and where is the endpoint on the rails end?

**Review by nikitr — CHANGES_REQUESTED** [review]

File: src/helpers/routes/routes.js (hunk 5 above)
Comment (clairelin135) [reply src/helpers/routes/routes.js:23]: o oops forgot to create PR for web, done now

**Review by nikitr — APPROVED** [review]
LGTM

**Pull request merged**